src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -158,7 +164,11 @@ def run_python_test(transitive_hydrated_target, pytest, python_setup, source_roo
 
   request = ExecuteProcessRequest(
     argv=(python_binary, './{}'.format(output_pytest_requirements_pex_filename)),
-    env={'PATH': os.pathsep.join(python_setup.interpreter_search_paths)},
+    env={
+      'PATH': os.pathsep.join(python_setup.interpreter_search_paths),
+      'LANG': os.environ.get('LANG', ''),
```

Comment:
Should we default to something like `en_US.UTF-8`?

Note that this will only impact the test runner, not the locale used for commands like `./pants run`.

I would advocate no to the default. Pants assumes that the overall environment supports UTF-8—which we eagerly check in `pants_loader.py`. I think the onus should be on the system to provide the encoding, and that it would be surprising for us to set here but not in other places.

Reply:
This one feels more like something that we should be setting on the `Pytest` subsystem if we aren't already, and then using in both v1 and v2...?

Reply:
I'd be concerned about setting the value only for `Pytest` because then its behavior could diverge from the rest of Pants. Let's say a user sets their encoding to `utf-8` for Pytest, but leaves the system default for, let's say, `./pants run`. Then, their tests might show everything working, but the actual `./pants run` would behave differently.

Instead, I think that the right decision is to require the external environment to be UTF-8, which we set the precedent for in `pants_loader.py`:

https://github.com/pantsbuild/pants/blob/c6be88522d4697e2df553be5f9893343076c74d6/src/python/pants/bin/pants_loader.py#L44-L61

Reply:
(Oops, hit send too early)

I think we might also want to consider having `ExecuteProcessRequest` _always_ grab the system values for `LC_ALL` and `LANG`. I cannot think of a scenario where we would not want to use the values. I'd be concerned a Pants developer might not know when they need to set the values and when they don't. Or, even worse, that they would forget to set it, the tests would not include testing for Unicode support, and then they ship code that breaks with Unicode without realizing it.

I know that we would like the environment to be as hermetic as possible, but I suspect there will be 2-3 values like this that are core assumptions to get Python behaving properly.

Reply:
I think capturing the local `LC_ALL` and `LANG` by default in the process execution system is probably reasonable. We should probably do this on the Python side in the `ExecuteProcessRequest` constructor (and in the future we could maybe add a config value to override them)